### PR TITLE
Fix code generation for conditions

### PR DESCRIPTION
**Pull request by veerajsk**

### CODE REVIEW

File: supplementary/alica_ros1/supplementary_tests/src/test/Expr/src/Behaviour/waitbehaviour1234.cpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#include <memory>
+#include <supplementary_tests/Behaviour/waitbehaviour1234.h>
```
Comment (veerajsk) [inline supplementary/alica_ros1/supplementary_tests/src/test/Expr/src/Behaviour/waitbehaviour1234.cpp:2]: @triccyx which behaviour is this? Seems to not be used anywhere, but the beh file is still around, so code gen generated code for it as well

Comment (Dale-Koenig) [reply supplementary/alica_ros1/supplementary_tests/src/test/Expr/src/Behaviour/waitbehaviour1234.cpp:2]: If it's not used just remove it

Comment (veerajsk) [reply supplementary/alica_ros1/supplementary_tests/src/test/Expr/src/Behaviour/waitbehaviour1234.cpp:2]: oops my bad. This is an autogenerated file for the waitbehaviour. Its id is 1234 becuase @triccyx manually added the beh file. @triccyx please use the plan designer for adding plans/behs & also use the code generation for generating the auto generated code.

Comment (veerajsk) [reply supplementary/alica_ros1/supplementary_tests/src/test/Expr/src/Behaviour/waitbehaviour1234.cpp:2]: I think you have added acme* behs, plans & conds manually & they are failing to import into the plan designer

Comment (triccyx) [reply supplementary/alica_ros1/supplementary_tests/src/test/Expr/src/Behaviour/waitbehaviour1234.cpp:2]: Ok understood.

**Review by triccyx — APPROVED** [review]

**Review by dhananjay-patki — APPROVED** [review]
Not the most familiar with how the code generation works, but I don't see any problems

File: supplementary/alica_ros1/supplementary_tests/src/test/Expr/include/supplementary_tests/Behaviour/waitbehaviour.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#pragma once
```
Comment (Dale-Koenig) [inline supplementary/alica_ros1/supplementary_tests/src/test/Expr/include/supplementary_tests/Behaviour/waitbehaviour.h:1]: This behavior doesn't do anything, remove it.

**Review by Dale-Koenig — APPROVED** [review]
I think I need to be explained why/how code generation is needed for dynamically loaded libraries to begin with

File: supplementary/alica_ros1/supplementary_tests/src/test/Expr/include/supplementary_tests/Behaviour/waitbehaviour.h (hunk 2 above)
Comment (veerajsk) [reply supplementary/alica_ros1/supplementary_tests/src/test/Expr/include/supplementary_tests/Behaviour/waitbehaviour.h:1]: This was generated automatically by the codegen. The actual problem is that plans & behs that are part of a dynamic library have their pml & beh files in the root etc folder & so it is causing problems during code gen. I will fix this in a subsequent PR

**Pull request merged**